crates/precompile/build.rs

```diff
@@ -0,0 +1,183 @@
+}
+
+/// Pros:
+/// - smaller runtime static size (198K = `4096*48 + 65*96`)
+/// - smaller generated file size (~~720K~~ same as runtime size by using `.bin` files)
+/// - half the size of `trusted_setup.txt`, which would have to be included in the binary without
+///   this build script
+///
+/// Cons:
+/// - must call `load_trusted_setup` at least once
+fn format_kzg_settings_small(in_path: &Path, out_dir: &Path) -> String {
+    let contents = fs::read_to_string(in_path).unwrap();
+    let mut lines = contents.lines();
+
+    // load number of points
+    let n_g1 = lines.next().unwrap().parse::<usize>().unwrap();
+    let n_g2 = lines.next().unwrap().parse::<usize>().unwrap();
+
+    assert_eq!(n_g2, 65);
+
+    // TODO: We can use `hex_literal` to reduce file size
+    // load g1 points
+    let mut g1_points = Vec::with_capacity(n_g1);
+    for _ in 0..n_g1 {
+        let line = lines.next().unwrap();
+        let mut bytes = [0; BYTES_PER_G1_POINT];
+        hex::decode_to_slice(line, &mut bytes).unwrap();
+        g1_points.push(bytes);
+    }
+
+    // load g2 points
+    let mut g2_points = Vec::with_capacity(n_g2);
+    for _ in 0..n_g2 {
+        let line = lines.next().unwrap();
+        let mut bytes = [0; BYTES_PER_G2_POINT];
+        hex::decode_to_slice(line, &mut bytes).unwrap();
+        g2_points.push(bytes);
+    }
+
+    assert!(lines.next().is_none());
+
+    fs::write(out_dir.join("g1_points.bin"), into_flattened(g1_points)).unwrap();
+    fs::write(out_dir.join("g2_points.bin"), into_flattened(g2_points)).unwrap();
+
+    format!(
+        r#"// @generated by build.rs from {in_path:?}, do not modify manually.
+
+pub use c_kzg::{{BYTES_PER_G1_POINT, BYTES_PER_G2_POINT}};
+
+// Ensure that the build script constants are synced with the C bindings ones.
+const _: [(); BYTES_PER_G1_POINT] = [(); {BYTES_PER_G1_POINT}];
+const _: [(); BYTES_PER_G2_POINT] = [(); {BYTES_PER_G2_POINT}];
+
+pub const NUM_G1_POINTS: usize = {n_g1};
+pub const NUM_G2_POINTS: usize = {n_g2};
+
+type G1Points = [[u8; BYTES_PER_G1_POINT]; NUM_G1_POINTS]; 
+type G2Points = [[u8; BYTES_PER_G2_POINT]; NUM_G2_POINTS]; 
+
+pub const G1_POINTS: &G1Points = {{
+    const BYTES: &[u8] = include_bytes!("./g1_points.bin");
+    assert!(BYTES.len() == core::mem::size_of::<G1Points>());
+    unsafe {{ &*BYTES.as_ptr().cast::<G1Points>() }}
+}};
+pub const G2_POINTS: &G2Points = {{
+    const BYTES: &[u8] = include_bytes!("./g2_points.bin");
+    assert!(BYTES.len() == core::mem::size_of::<G2Points>());
+    unsafe {{ &*BYTES.as_ptr().cast::<G2Points>() }}
+}};
+"#
+    )
+}
+
+/*
+/// Pros:
+/// - no need to call `load_trusted_setup` at runtime
+///
+/// Cons:
+/// - larger runtime static size (722K = `4096*32 + 4096*144 + 65*288`)
+/// - larger generated file size (2M)
+/// - possibly unsafe? `as_ptr() as *mut _`
+/// - depends on `c_kzg`; this might not matter much
+fn format_kzg_settings_large(in_path: &Path) -> String {
```

Comment:
We should check about this w/ the c-kzg folks.

Reply:
I think I'll delete the second option, it's way too big to have at runtime